pallets/initializer/src/lib.rs

```diff
@@ -139,14 +95,7 @@ impl<T: Config> Pallet<T> {
             // Genesis session should be immediately enacted.
             T::SessionHandler::apply_new_session(false, 0u32.into(), validators, queued);
         } else {
-            BufferedSessionChanges::<T>::mutate(|v| {
-                *v = Some(BufferedSessionChange {
-                    changed,
-                    validators,
-                    queued,
-                    session_index,
-                })
-            });
+            T::SessionHandler::apply_new_session(changed, session_index, validators, queued);
```

Comment:
This if-else is now no longer needed. I assume changed will be false when session_index == 0, so you can just always do:

```rust
T::SessionHandler::apply_new_session(changed, session_index, validators, queued);
```